shared/src/packages/drivers/libs/enums/driver-validation-rule.enum.ts

```diff
@@ -0,0 +1,8 @@
+const DriverValidationRule = {
+  LICENSE_MIN_LENGTH: 5,
+  LICENSE_MAX_LENGTH: 30,
+  LICENSE_NUMBER: /^[\dA-Z]+(?:[ -][\dA-Z]+)*$/,
```

Comment:
```suggestion
  LICENSE_NUMBER: /^(?=.{5,30}$)[\dA-Z]+(?:[ -][\dA-Z]+)*$/,
```

Reply:
This would count all symbols. I'm pretty sure, that min 5 max 30 constraint mean only meaningful symbols, like letters and numbers. This does not include hyphens (-) and whitespaces. Otherwise what is the point of restriction, if the same number can be written as 1-2-3-4-5(9 characters, but 5 meaningful symbols) or 12345 (5 characters and 5 meaningful symbols).

I'm gonna leave it like this by now unless somebody prove me I'm wrong.

Reply:
What will be stored in the database? Value with spaces or without them? I'm not sure we want to store things like this `FOO1 2 3 45 A A A A A A A A A A A A A A A A A A A A A A`

Reply:
It looks strange to use regex in an enum that has fields with string length constraints. You could include it directly in the regex.

Reply:
So what are you suggestions? there is no defined template for number. All we know, that it is possible to have numbers, letters, hyphens and spaces. The template i use checks exactly this, with constraint that you cannot use combination of double hyphen/spaces.

Speaking on checking the length, though, it is restricted to 5...30 MEANINGFUL characters (letters and symbols).

If you want me to prohibit hyphens and spaces, i'm okay about it. If you want me to count hyphens and spaces as meaningful letters, i'm not fan of that solution, because this makes no sense to restrict the value that way to me.

Reply:
@PavloSukhinin there are certain formats of writing license plates, they may or may not include hyphens and spaces. 
If you want it so, i can strip all those delimeters and store a pure meaningful value in the db. But it will lack proper definition. Not sure, if it is okay to have AB123456 instead of AB 123-456 (if the country respects this precise format)

Reply:
>It looks strange to use regex in an enum that has fields with string length constraints. You could include it directly in the regex.
@denred i'm quite opposite to this
do  you think it is okay to hide the length constraints in something like this : 
```
/^(?=.{5,30}$)[\dA-Z]+(?:[ -][\dA-Z]+)*$/,
(find the limits in 10 seconds :) )

```
vs this

```
  LICENSE_MIN_LENGTH: 5,
  LICENSE_MAX_LENGTH: 30,
  LICENSE_NUMBER: /^[\dA-Z]+(?:[ -][\dA-Z]+)*$/,
```

Reply:
I meant that it would be a good idea to move the regex into a separate file or variable, rather than keeping it in the enum. It would also be good to specify the limits of the license number directly in the regex and extract the length parameters into an enum as you already have.